extensions/email/src/main/java/org/mule/extension/email/api/EmailAttributes.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.api;
+
+import org.mule.extension.email.internal.EmailFlags;
+import org.mule.runtime.api.message.MuleMessage;
+
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.activation.DataHandler;
+import javax.mail.Folder;
+import javax.mail.Message;
+
+/**
+ * Contains all the metadata of an email, it carries information such as
+ * the subject of the email, the id in the mailbox and the recipitiens between others.
+ * <p>
+ * This class aims to be returned as attributes in a {@link MuleMessage} for every
+ * retriever operation.
+ * <p>
+ * The attachments of the email are also carried in an {@link EmailAttributes} instance
+ * and separated from the original multipart {@link Message}.
+ *
+ * @since 4.0
+ */
+public class EmailAttributes implements Serializable
+{
+
+    private int id;
+    private List<String> fromAddresses = new ArrayList<>();
```

Comment:
looks like the javadocs that I'm talking about you actually put on the methods.. Put on the fields since that's what the SDK will introspect